keymanager/oci_key_manager.go

```diff
@@ -0,0 +1,103 @@
+/*
+ *   Copyright (c) 2024 Oracle Corporation
+ *   All rights reserved.
+ *   SPDX-License-Identifier: BSD-3-Clause
+ */
+
+package keymanager
+
+import (
+	"intel/kbs/v1/model"
+	"intel/kbs/v1/ociclient"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type OCIManager struct {
+	client ociclient.OCIClient
+}
+
+func NewOCIManager(c ociclient.OCIClient) *OCIManager {
+	return &OCIManager{c}
+}
+
+func (om *OCIManager) CreateKey(keyRequest *model.KeyRequest) (*model.KeyAttributes, error) {
+	if keyRequest.KeyInfo.OciCompartmentId == "" || keyRequest.KeyInfo.OciKeyId == "" ||
+		keyRequest.KeyInfo.OciSecretName == "" || keyRequest.KeyInfo.OciVaultId == "" {
+		return nil, errors.New("Missing oci_compartment_id, oci_key_id, oci_secret_name, or oci_vault_id")
+	}
+
+	newUuid, err := uuid.NewRandom()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create new UUID")
+	}
+	keyAttributes := &model.KeyAttributes{
+		ID:               newUuid,
+		Algorithm:        keyRequest.KeyInfo.Algorithm,
+		KeyLength:        keyRequest.KeyInfo.KeyLength,
+		OciCompartmentId: keyRequest.KeyInfo.OciCompartmentId,
+		OciKeyId:         keyRequest.KeyInfo.OciKeyId,
+		OciSecretName:    keyRequest.KeyInfo.OciSecretName,
+		OciVaultId:       keyRequest.KeyInfo.OciVaultId,
+		TransferPolicyId: keyRequest.TransferPolicyID,
+		CreatedAt:        time.Now().UTC(),
+	}
+
+	log.Infof("OCI: Creating key: algorithm = %q; secret name = %q", keyAttributes.Algorithm, keyAttributes.OciSecretName)
+
+	if err := om.client.CreateKey(keyAttributes); err != nil {
+		return nil, errors.Wrap(err, "failed to create key")
+	}
+
+	return keyAttributes, nil
+}
+
+func (om *OCIManager) DeleteKey(keyAttributes *model.KeyAttributes) error {
+	err := om.client.DeleteKey(keyAttributes.OciSecretId)
+	if err != nil {
+		log.Errorf("Error while deleting key: %s", err.Error())
```

Comment:
I'd suggest to re-consider the usage of `github.com/pkg/errors` since it's archived and not maintained. In general, projects are trying to move away from it and use error wrapping from the std library.

Reply:
I'll update it to errors.Wrap() for now. I was just copying the code from   vault_key_manager.go:DeleteKey(). If we want to move the std library implementation, I would suggest doing that as a separate patch(set) that updates error usage everywhere.

I'm also not 100% sure what wrapping from the std library you want to do -- I don't see an errors.Wrap() function there. Do you mean using fmt.Errorf() or errors.Join()?

Reply:
I was thinking`%w`  with `err`